lms/tasks/email_digests.py

```diff
@@ -93,6 +99,15 @@ def send_instructor_email_digests(
     with app.request_context() as request:  # pylint:disable=no-member
         with request.tm:
             digest_service = request.find_service(DigestService)
+            email_unsubscribe_service = request.find_service(EmailUnsubscribeService)
+
+            h_userids = email_unsubscribe_service.filter_unsubscribed_h_userids(
+                EmailUnsubscribe.Tag.INSTRUCTOR_DIGEST, h_userids
+            )
+            if not h_userids:
```

Comment:
Not likely (hope so) in prod but hit this during testing.

Reply:
I still think that `send_instructor_email_digest_tasks()` is more the right place to do this filtering: it won't mess up the batching like this, and it's where the other filtering is done. For example it filters out users who don't belong to an application instance with the feature flag enabled. In the future we may extend `send_instructor_email_digest_tasks()`'s DB query to also filter out users who don't belong to a course with an assignment that has been launched in the last 24 hours (since we know they can't have any activity).

I know that means that the admin page will ignore unsubscribes but I don't think we actually need the admin page to test unsubscribes. The admin page also ignores the feature flag: it'll send you an email regardless of whether the user's application instance has the feature enabled or not.

If you wanted to test it manually you could run `make shell` and then run `send_instructor_email_digest_tasks.delay()`. Admittedly this only works in dev not on QA or prod but I think it's fine.